src/services/VerifyAccountRequestProcessor.ts

```diff
@@ -0,0 +1,92 @@
+
+export class VerifyAccountRequestProcessor {
+  private static instance: VerifyAccountRequestProcessor;
+
+  private readonly logger: Logger;
+
+	private readonly personIdentityTableName: string;
+
+  private readonly metrics: Metrics;
+
+  private readonly BavService: BavService;
+	
+  private readonly HmrcService: HmrcService;
+
+	private readonly hmrcTokenSsmPath: string;
+
+	constructor(logger: Logger, metrics: Metrics) {
+  	this.logger = logger;
+  	this.metrics = metrics;
+  	logger.debug("metrics is  " + JSON.stringify(this.metrics));
+  	this.metrics.addMetric("Called", MetricUnits.Count, 1);
+
+  	const sessionTableName: string = checkEnvironmentVariable(EnvironmentVariables.SESSION_TABLE, this.logger);
+  	this.personIdentityTableName = checkEnvironmentVariable(EnvironmentVariables.PERSON_IDENTITY_TABLE_NAME, this.logger);
+  	const hmrcBaseUrl = checkEnvironmentVariable(EnvironmentVariables.HMRC_BASE_URL, this.logger);
+  	this.hmrcTokenSsmPath = checkEnvironmentVariable(EnvironmentVariables.HMRC_TOKEN_SSM_PATH, this.logger);
+
+  	this.BavService = BavService.getInstance(sessionTableName, this.logger, createDynamoDbClient());
+  	this.HmrcService = HmrcService.getInstance(this.logger, hmrcBaseUrl);
+	}
+
+	static getInstance(logger: Logger, metrics: Metrics): VerifyAccountRequestProcessor {
+  	if (!VerifyAccountRequestProcessor.instance) {
+  		VerifyAccountRequestProcessor.instance = new VerifyAccountRequestProcessor(logger, metrics);
+  	}
+  	return VerifyAccountRequestProcessor.instance;
+	}
+
+	async processRequest(sessionId: string, body: VerifyAccountPayload): Promise<Response> {
+  	const { account_number: accountNumber, sort_code: sortCode } = body;
+		const paddedAccountNumber = accountNumber.padStart(8, "0");
+  	const person: PersonIdentityItem | undefined = await this.BavService.getPersonIdentityById(sessionId, this.personIdentityTableName);
+
+  	if (!person) {
+  		this.logger.error("No person found for session id", {
+  			messageCode: MessageCodes.SESSION_NOT_FOUND,
+  		});
+  		return new Response(HttpCodesEnum.UNAUTHORIZED, `No person found with the session id: ${sessionId}`);
+  	}
+
+  	await this.BavService.updateAccountDetails(sessionId, paddedAccountNumber, sortCode, this.personIdentityTableName);
+
+  	const name = getFullName(person.name);
+  	const verifyResponse = await this.HmrcService.verify({ accountNumber: paddedAccountNumber, sortCode, name }, this.hmrcTokenSsmPath);
+
+  	const copCheckResult = this.calculateCopCheckResult(verifyResponse);
+  	this.logger.debug(`copCheckResult is ${copCheckResult}`);
+
+  	await this.BavService.saveCopCheckResult(sessionId, copCheckResult);
+
+  	if (copCheckResult === CopCheckResults.NO_MATCH) {
+  		return new Response(HttpCodesEnum.SERVER_ERROR, "Verification failed");
+  	}
+  	return new Response(HttpCodesEnum.OK, "Success");
+	}
+
+	calculateCopCheckResult(verifyResponse: HmrcVerifyResponse): CopCheckResult {
```

Comment:
I know it's not in the ACs but I wonder if there's a benefit here of a separate check and a diff response should the account not exist? Just thinking if the user accidentally entered the account details incorrectly we don't wanna cost them a retry or worse a CI VC

Reply:
Yeah, I think the error scenarios are a lot more complex than we anticipated! Happy to cover this in the 5xx ticket?

Reply:
Yup :)